### PR TITLE
better handling of the directoryenv configuration

### PR DESCRIPTION
**Pull request by tampakrap**
add default value for environmentpath
Use default puppet values for manifest and modulepath
Remove any non-defined directoryenv related variables from puppet.conf,
otherwise we end up with a broken config
Rename config_version_cmd to config_version, to match the puppet.conf option

### CODE REVIEW

**Comment by tampakrap** [comment]
I don't understand the test failure

**Comment by tampakrap** [comment]
probably fixed with #208 although I still don't understand why puppet-lint failed on warnings instead of errors. We explicitly set in Rakefile to fail only on warnings

**Comment by tampakrap** [comment]
tests are green now, please review

**Comment by xaque208** [comment]
+1  Nice work.

File: manifests/server.pp [hunk 1]
```diff
@@ -27,35 +27,35 @@
...
-  $serverssl_ciphers  = undef,
-  $serverssl_protos   = undef,
-  $servertype         = 'unicorn',
-  $storeconfigs       = undef,
-  $stringify_facts    = false,
-  $package            = $puppet::params::master_package,
+  $autosign          = undef,
+  $bindaddress       = '0.0.0.0',
+  $ca                = false,
+  $config_version    = '/usr/bin/git --git-dir $confdir/environments/$environment/.git rev-parse --short HEAD 2>/dev/null || echo',
```
Comment (xaque208) [inline manifests/server.pp]: Doesn't this setting go away with directory environments?

Comment (tampakrap) [inline manifests/server.pp]: Yes, when `puppet::server::directoryenvs` is true, then the `manifest`, `modulepath` and `config_version` variables get removed from puppet.conf (lines 32-34 in `puppet::server::config` class)

**Pull request merged**